notebook/services/kernels/handlers.py

```diff
@@ -219,6 +219,7 @@ def pre_get(self):
         # We don't want to wait forever, because browsers don't take it well when
         # servers never respond to websocket connection requests.
         kernel = self.kernel_manager.get_kernel(self.kernel_id)
+        self._open_sessions[self.session_key] = self
```

Comment:
Rather than remove the initialization of the `_open_sessions` entry from `_register_session()` would it make sense to check the kernel_id's presence prior to `_register_session()` instead (and keep `_register_session()` intact)?

Reply:
@kevin-bates : `_register_session` is also responsible for cleaning up any existing / stale connection..so we can not skip the call completely by checking kernel's presence. 

We felt this solution is much cleaner as we populate the `_open_sessions`  after `get_kernel` which makes sure that the kernel exists.

Reply:
I see - good point since `get_kernel` throws if the id is not present - understood.  I would just rather not remove the one line that "registers" a session from `_register_session()`.  

What about checking the kernel manager's existence in a more passive manner within `_register_session()` and conditionally updating `_open_sessions`:
```python
        self.session_key = '%s:%s' % (self.kernel_id, self.session.session)
        stale_handler = self._open_sessions.get(self.session_key)
        if stale_handler:
            self.log.warning("Replacing stale connection: %s", self.session_key)
            yield stale_handler.close()
        if self.kernel_id in self.kernel_manager:  # only update open sessions if kernel is actively managed
            self._open_sessions[self.session_key] = self
```
while the subsequent `get_kernel()` call throws.

Reply:
+1 @kevin-bates. 
@Vishwajeet0510 : please test that adding this extra check does not have corner cases.

Reply:
Hello @kevin-bates, @rahul26goyal 

I've verified the above mentioned change and it is working fine, I will update the PR with this.

Thanks Kevin